streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala

```diff
@@ -20,20 +20,26 @@ package org.apache.spark.streaming.ui
 import java.util.Calendar
 import javax.servlet.http.HttpServletRequest
 
-import scala.xml.Node
-
 import org.apache.spark.Logging
-import org.apache.spark.ui._
 import org.apache.spark.ui.UIUtils._
+import org.apache.spark.ui._
 import org.apache.spark.util.Distribution
 
+import scala.xml.Node
+
 /** Page for Spark Web UI that shows statistics of a streaming job */
 private[ui] class StreamingPage(parent: StreamingTab)
   extends WebUIPage("") with Logging {
 
-  private val listener = parent.listener
-  private val startTime = Calendar.getInstance().getTime()
+  private var listener = parent.listener
+  private var startTime = Calendar.getInstance().getTime()
   private val emptyCell = "-"
+  
+  def updateListener(listener : StreamingJobProgressListener):
+  Unit ={
```

Comment:
The wrapping is wrong here; `Unit` should be on the previous line and no space before the brace. No space before `:`. Also, `System.currentTimeMillis()` is much more conventional for getting the time.